docs/en/advanced-concepts/android-testing-with-docker.md

```diff
@@ -0,0 +1,66 @@
+## Android Tests With Docker
+
+When time comes to run appium tests on your own CI (like Jenkins) you would probably think about using docker.
+In order to run Android tests you will also need to have an emulator. Fortunately you can find official docker image that has both Appium and Android Emulator bundled inside: 
+https://hub.docker.com/r/appium/appium-emulator/
+ 
+All you need to know is just 2 simple tricks:
+1. Run ADB server using `-a` switch in the docker container running emulator. ADB will listen to all connected network interfaces, so you can connect emulator from another running container
+2. You will need to setup port forwarding from the container running Appium to the container running emulator. 
+Appium uses a `bootstrap-port` to setup communication to the emulator and it will bind to the localhost by default.
+But since emulator running in another container, you will need to bind that local `bootstrap-port` to the same port on the emulator container
+
+Here is a sample docker compose file that does all these tricks:
+```yaml
+version: "2"
+
+services:
+  emulator:
+    container_name: emulator
+    image: appium/appium-emulator
+    command: >
+          bash -c "adb -a nodaemon server &
+                   emulator64-arm -avd Nexus -no-boot-anim -no-window -noaudio -gpu off"
+
+  appium:
+    container_name: appium
+    image: appium/appium-emulator
```

Comment:
@serggl You can check https://github.com/butomo1989/docker-android. If your CI is linux machine, you can use x86 system image for the fastest performance. for x86 system image it takes around 10-40 sec and arm system image it takes around 10 min like you mentioned (all depends on computer specification).  
and we run appium server in separate thread, so you don't need to run appium server separately in different container. In case you want to combine all containers in one hub, you can use selenium hub.